lib/ruby_http_client.rb

```diff
@@ -184,8 +184,8 @@ def build_args(args)
     #   - The final url string
     #
     def build_url(query_params: nil)
-      url = [add_version(''), *@url_path].join('/')
-      url = build_query_params(url, query_params) if query_params
+      url = [add_version(+''), *@url_path].join('/')
+      url = build_query_params(+url, query_params) if query_params
```

Comment:
I don't think this `url` is frozen because it is not “string literal” 🤔
```suggestion
      url = build_query_params(url, query_params) if query_params
```

Reply:
It is.  The line before does a join('/')

So it creates a string literal